pf9-express

```diff
@@ -618,9 +619,18 @@ proxy_url=$(grep ^proxy_url ${pf9_config} | cut -d \| -f2)
 ## append proxy_url to extra_args
 if [ "${proxy_url}" != "-" ]; then extra_vars="${extra_vars} proxy_url=${proxy_url}"; fi
 
+## get region specific hostname
+## Get Keystone Token
+token=`curl -k -i -H "Content-Type: application/json" https://${du_url}/keystone/v3/auth/tokens?nocatalog \
+    -d "{ \"auth\": { \"identity\": { \"methods\": [\"password\"], \"password\": { \"user\": { \"name\": \"${du_username}\", \"domain\": {\"id\": \"default\"}, \"password\": \"${du_password}\" } } }, \"scope\": { \"project\": { \"name\": \"${du_tenant}\", \"domain\": {\"id\": \"default\"}}}}}" 2>/dev/null | grep -i ^X-Subject-Token | awk -F : '{print $2}' | sed -e 's/ //g' | sed -e 's/\r//g'`
+
+## get region info
+service_id=`curl -X GET -H "X-Auth-Token: ${token}" https://${du_url}/keystone/v3/services?type=regionInfo 2>/dev/null | grep -o '"id": "[a-z0-9 ]*' | cut -c 8-40`
+region_url=`curl -X GET -H "X-Auth-Token: ${token}" https://${du_url}/keystone/v3/endpoints?service_id=${service_id} | jq --arg du_region "${du_region}" '.endpoints[] | select(.region_id==$du_region and .interface=="public")' | jq '.url' | awk '{print substr($0,2,length($0)-9)}' | cut -d '/' -f3`
+
 ## assign/validate ctrl_ip from config file and resolve IP for ctrl_hostname
-ctrl_hostname=$(grep ^du_url ${pf9_config} | cut -d \| -f2 | cut -d \/ -f3)
-tmp_ip=$(ping -c 1 ${ctrl_hostname} | grep PING | cut -d ' ' -f3)
+ctrl_hostname=$( ${du_url} | cut -d \| -f2 | cut -d \/ -f3)
```

Comment:
This:
ctrl_hostname=$( ${du_url} | cut -d \| -f2 | cut -d \/ -f3)

Should read:
ctrl_hostname=$(echo ${du_url} | cut -d \| -f2 | cut -d \/ -f3)